src/adapter/src/optimize/materialized_view.rs

```diff
@@ -82,6 +83,8 @@ pub struct Optimizer {
     ///
     /// Used to determine if it is safe to enable dataflow expiration.
     is_timeline_epoch_ms: bool,
+    /// Overrides monotonicity for the given collections.
+    force_non_monotonic: BTreeSet<GlobalId>,
```

Comment:
How about adding a note about why we need this and to remove this when/if we separate the CT optimizer from the MV optimizer?

Reply:
Also this only forces non-monotonicity for source inputs, not index inputs, can we put a `source` somewhere in the name to make that clear?